stock_location_bin_name/tests/test_stock_location_bin_name.py

```diff
@@ -0,0 +1,137 @@
+from odoo.tests import Form
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+class TestStockLocationBinName(BaseCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.location_obj = cls.env["stock.location"]
+        # Create the following structure:
+        # # [Zone Location]
+        # # # [Area Location]
+        # # # # [Bin Location]
+        cls.location_zone = cls.location_obj.create(
+            {
+                "name": "Zone Location",
+                "is_zone": True,
+            }
+        )
+        cls.location_area = cls.location_obj.create(
+            {"name": "Area Location", "location_id": cls.location_zone.id}
+        )
+
+        cls.location_bin = cls.location_obj.create(
+            {
+                "name": "Bin Location",
+                "zone_location_id": cls.location_zone.id,
+                "area_location_id": cls.location_area.id,
+                "location_id": cls.location_area.id,
+                "location_kind": "bin",
+                "usage": "internal",
+            }
+        )
+
+    def test_location_name_format(self):
+        """Test location name with format string from area"""
+
+        with Form(self.location_area) as location_form:
+            location_form.location_name_format = (
+                "{area}-{corridor}-{posx:0>3}.{rack:0>3}.{level:0>2}"
+            )
+
+        with Form(self.location_bin) as location_form:
+            location_form.corridor = "A"
+            location_form.rack = "51"
+            location_form.posx = 123
+            location_form.level = "42"
+
+        self.assertEqual(self.location_bin.name, "Area Location-A-123.051.42")
+
+    def test_location_name_format_in_zone(self):
+        """Test location name with format string from zone"""
+
+        with Form(self.location_zone) as location_form:
+            location_form.location_name_format = (
+                "{area}-{corridor}-{level:0>2}-{posx}{posy}{posz}.{rack:0>3}"
+            )
+
+        with Form(self.location_bin) as location_form:
+            location_form.corridor = "Z"
+            location_form.rack = "5"
+            location_form.posx = 1
+            location_form.posy = 2
+            location_form.posz = 3
+            location_form.level = "2"
+
+        self.assertEqual(self.location_bin.name, "Zone Location-Z-02-123.005")
+
+    def test_location_bin_with_name_format(self):
+        """Test location constraint with location name format in a bin location"""
+
+        with self.assertRaises(Exception) as context:
```

Comment:
a little tip here:
When testing an exception, we want to ensure that the raised exception is actually the one we introduced.
For that, we should use `assertRaisesRegex` instead:
```suggestion
regex = "Location Name Format cannot be set on 'bin' type locations."
with self.assertRaisesRegex(ValidationError, regex):
```
Then you don't need to check the context :blush: